### ISSUE TITLE
Windows support in eventloop.d?

### ISSUE THREAD

**Issue by quickfur** [adamdruppe/arsd#230]
We've already exchanged emails on this some time ago, but just putting this ticket here so that it doesn't get forgotten.

Basically all I need is some kind of event loop support for terminal.d that I can use for my program's main loop. It doesn't have to be eventloop.d specifically, as long as there's some way of hooking terminal.d into an event loop of some kind. Could be hooks for writing my own code, though it would be nicer if there was a cross-platform solution.

**Comment by adamdruppe** [comment]
Yes, indeed, i actually looked at the emails just a couple days ago.

I'll just have to find the time, i have 6 other active things right now
lol

**Comment by quickfur** [comment]
Just a side-note: in my current code I've managed to find an easy workaround for now: since the code hasn't actually reached the point where I need to integrate other events into the main loop, I got away with just getting rid of the `arsd.eventloop` dependency entirely and replacing the main loop with:

````
auto input = RealTimeConsoleInput(...);
...
while (!quit) {
    auto ev = input.nextEvent();
    myEventHandler(ev);
}
````

This lets me compile the same code for Linux and Windows and have it essentially work the same.

When it comes time to add other types of events I think I'll probably just implement my own `select` inside the loop body and delegate to `input.nextEvent` only when stdin is ready for reading. Not sure what I'd do in the Windows case, though, since `select` is Posix-specific.  Will cross that bridge when I get there. :-D

Anyway, the tl;dr is that this issue isn't as high of a priority for me anymore, even though it's still nice to have.

**Comment by adamdruppe** [comment]
Yeah, that's how I do it in the demo, but integration with other stuff is tricky.

I just need to find the time to do it all.
